cmd/network-slirp-binding/domain/domain.go

```diff
@@ -0,0 +1,175 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package domain
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	vmschema "kubevirt.io/api/core/v1"
+
+	domainschema "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
+type SlirpNetworkConfigurator struct {
+	vmiSpecIface   *vmschema.Interface
+	vmiSpecNetwork *vmschema.Network
+	searchDomains  []string
+}
+
+func NewSlirpNetworkConfigurator(ifaces []vmschema.Interface, networks []vmschema.Network, searchDomains []string) (*SlirpNetworkConfigurator, error) {
+	network := vmispec.LookupPodNetwork(networks)
+	if network == nil {
+		return nil, fmt.Errorf("pod network not found")
+	}
+	iface := vmispec.LookupInterfaceByName(ifaces, network.Name)
+	if iface == nil || iface.Slirp == nil {
+		return nil, fmt.Errorf("no slirp iface found")
+	}
+
+	return &SlirpNetworkConfigurator{
+		vmiSpecIface:   iface,
+		vmiSpecNetwork: network,
+		searchDomains:  searchDomains,
+	}, nil
+}
+
+func (s SlirpNetworkConfigurator) Mutate(domainSpec *domainschema.DomainSpec) (*domainschema.DomainSpec, error) {
+	slirpQemuCmdArgs, err := generateSlirpNetworkQemuCmdArgs(s.vmiSpecIface, s.vmiSpecNetwork, s.searchDomains)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(slirpQemuCmdArgs) == 0 {
+		log.Log.Warning("no qemu cmd args configured, domain spec did not change")
+		return domainSpec, nil
+	}
+
+	domainSpec.QEMUCmd = initQemuCmdArgs(domainSpec.QEMUCmd)
```

Comment:
The converter does [domainIface.Type = "user"](https://github.com/kubevirt/kubevirt/blob/b533699f568f895a7960a0735dfdda20dfcb0424/pkg/virt-launcher/virtwrap/converter/network.go#L109), is it redundant?

Reply:
Yes, the converter adds an interfaces to the domain spec, which get deleted by network-setup phase2 code
https://github.com/kubevirt/kubevirt/blob/main/pkg/network/domainspec/generators.go#L218